src/openvic-simulation/economy/FactoryProducer.cpp

```diff
@@ -37,3 +44,71 @@ fixed_point_t FactoryProducer::get_average_profitability_last_seven_days() const
 
 	return sum / (1 + profit_history_current);
 }
+
+void FactoryProducer::produce(
+	const fixed_point_t input_modifier, const fixed_point_t throughput_modifier, const fixed_point_t output_modifier
+) {
+	std::map<PopType const* const, fixed_point_t> employees_per_job {};
+	for (const Job job : production_type.get_jobs()) {
```

Comment:
```suggestion
	for (Job const& job : production_type.get_jobs()) {
```
Every time a non-pointer or reference variable is created, it's entire contents is copied to a new block of memory. Better to use references and avoid creating unnecessary intermediate variables. Same for the other `for` loops in this file.